modules/library/http/src/main/java/org/geotools/http/AbstractHttpClient.java

```diff
@@ -90,6 +108,33 @@ public boolean isTryGzip() {
         return tryGzip;
     }
 
+    protected static URL appendURL(URL oldUrl, Map<String, Object> appendQuery)
```

Comment:
We need a javadoc here as this ends up being a semi-public method due to protected visibility.